cmd/agent/app/reporter/tchannel/collector_proxy.go

```diff
@@ -25,22 +25,28 @@ import (
 
 // ProxyBuilder holds objects communicating with collector
 type ProxyBuilder struct {
-	reporter reporter.Reporter
+	reporter *reporter.ClientMetricsReporter
 	manager  configmanager.ClientConfigManager
 	tchanRep *Reporter
 }
 
 // NewCollectorProxy creates ProxyBuilder
 func NewCollectorProxy(builder *Builder, mFactory metrics.Factory, logger *zap.Logger) (*ProxyBuilder, error) {
-	r, err := builder.CreateReporter(logger)
+	tchanRep, err := builder.CreateReporter(logger)
 	if err != nil {
 		return nil, err
 	}
-	tchannelMetrics := mFactory.Namespace(metrics.NSOptions{Name: "", Tags: map[string]string{"protocol": "tchannel"}})
+	tchanMetrics := mFactory.Namespace(metrics.NSOptions{Name: "", Tags: map[string]string{"protocol": "tchannel"}})
+	r1 := reporter.WrapWithMetrics(tchanRep, tchanMetrics)
+	r2 := reporter.WrapWithClientMetrics(reporter.ClientMetricsReporterParams{
+		Reporter:       r1,
+		Logger:         logger,
+		MetricsFactory: mFactory,
+	})
 	return &ProxyBuilder{
-		tchanRep: r,
-		reporter: reporter.WrapWithMetrics(r, tchannelMetrics),
-		manager:  configmanager.WrapWithMetrics(tchannel.NewConfigManager(r.CollectorServiceName(), r.Channel()), tchannelMetrics),
+		tchanRep: tchanRep,
```

Comment:
Why do we need to export tchanRep? Is tchanRep used any where? If not, we could remove tchanRep from below struct:
`type ProxyBuilder struct {
	reporter reporter.Reporter
	manager  configmanager.ClientConfigManager
	tchanRep *Reporter
}`

Reply:
Yes, it's unfortunate side effect of the way constructors are defined. I think the Sampling server depends on the underlying tchannel that is being created here.